packages/studio-plugin/src/orchestrators/ParsingOrchestrator.ts

```diff
@@ -50,17 +50,22 @@ export default class ParsingOrchestrator {
     this.paths = studioConfig.paths;
     this.filepathToFileMetadata = this.initFilepathToFileMetadata();
     this.pageNameToPageFile = this.initPageNameToPageFile();
+    this.layoutOrchestrator = new LayoutOrchestrator(
+      this.paths,
+      this.project,
+      this.getFileMetadata
+    );
   }
 
-  getPageFile(pageName: string): PageFile {
+  getOrCreatePageFile = (pageName: string): PageFile => {
     const pageFile = this.pageNameToPageFile[pageName];
     if (pageFile) {
       return pageFile;
     }
     return this.createPageFile(pageName);
-  }
+  };
 
-  private createPageFile(pageName: string) {
+  private createPageFile = (pageName: string) => {
```

Comment:
Just our of curiosity, what is the advantage of having this as an arrow function as opposed to a standard function declaration?

Reply:
It has to do with `this` binding (see [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this)). Unlike standard functions, arrow functions don't bring their own `this` binding, so `this` retains the value from the calling function. Since we're passing `createPageFile` as a parameter to another function, we want it to retain `this` instead of creating a new binding for it, which would result in errors about accessing properties of undefined anywhere we do `this.blah` within the body of `createPageFile`. To do this, we can either change the syntax to an arrow function like we did here, or use the `bind` [method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#the_bind_method)

Reply:
oh ok, I see-- thanks for the explanation!